### PR TITLE
Remove Legacy Unused Code

### PR DESCRIPTION
**Pull request by brandonlehmann**
* Remove testnet references that have not been supported for some time.
* Remove premine references that have **never** been used during the **entire** life of the project.

### CODE REVIEW

File: src/Daemon/Daemon.cpp [hunk 1]
```diff
@@ -50,41 +50,16 @@ using namespace CryptoNote;
...
-    CryptoNote::AccountPublicAddress address;
-    if (!currency.parseAccountAddressString(rewardAddress, address))
-    {
-      std::cout << "Failed to parse genesis reward address: " << rewardAddress << std::endl;
-      return;
-    }
-    rewardTargets.emplace_back(std::move(address));
-  }
-
   CryptoNote::Transaction transaction;
```
Comment (zpalmtree) [inline src/Daemon/Daemon.cpp]: Can just do
```cpp
const CryptoNote::Transaction transaction = CryptoNote::CurrencyBuilder(logManager).generateGenesisTransaction();
```
or
```cpp
const auto transaction = CryptoNote::CurrencyBuilder(logManager).generateGenesisTransaction();
```
if that's your thing, now we don't need to conditionally assign it

File: src/P2p/NetNode.cpp [hunk 2]
```diff
@@ -380,12 +380,8 @@ std::string print_peerlist_to_string(const std::list<PeerlistEntry>& pl) {
   //-----------------------------------------------------------------------------------
 
   bool NodeServer::init(const NetNodeConfig& config) {
-    if (!config.getTestnet()) {
-      for (auto seed : CryptoNote::SEED_NODES) {
-        append_net_address(m_seed_nodes, seed);
-      }
-    } else {
-      m_network_id.data[0] += 1;
+    for (auto seed : CryptoNote::SEED_NODES) {
```
Comment (zpalmtree) [inline src/P2p/NetNode.cpp]:
```suggestion
    for (const auto &seed : CryptoNote::SEED_NODES) {
```

File: src/P2p/P2pNodeConfig.cpp [hunk 3]
```diff
@@ -1,19 +1,7 @@
 // Copyright (c) 2012-2017, The CryptoNote developers, The Bytecoin developers
+// Copyright (c) 2018, The TurtleCoin Developers
```
Comment (zpalmtree) [inline src/P2p/P2pNodeConfig.cpp]:
```suggestion
// Copyright (c) 2018-2019, The TurtleCoin Developers
```

File: src/Rpc/CoreRpcServerCommandsDefinitions.h [hunk 4]
```diff
@@ -358,7 +357,6 @@ struct COMMAND_RPC_GET_INFO {
       KV_MEMBER(minor_version)
       KV_MEMBER(start_time)
       KV_MEMBER(synced)
-      KV_MEMBER(testnet)
```
Comment (zpalmtree) [inline src/Rpc/CoreRpcServerCommandsDefinitions.h]: Technically we're changing the API here, no? Really doubt anyone is using it, but worth noting. What are your thoughts?

Comment (brandonlehmann) [reply src/Rpc/CoreRpcServerCommandsDefinitions.h]: Given that the `--testnet` flag hasn't existed and thus this was never set to true, any front-end utilities will see this as `false` when checking for it anyways as I don't recall ever coming across anything the explicitly checked for a an actual type `false`. It's minor, and we can not it in the release notes but I don't think it's going to hurt anything.

File: src/Daemon/Daemon.cpp (hunk 1 above)
Comment (brandonlehmann) [reply src/Daemon/Daemon.cpp]: Updated. I missed that while ripping and cleaning a bit.

File: src/Rpc/CoreRpcServerCommandsDefinitions.h (hunk 4 above)
Comment (zpalmtree) [reply src/Rpc/CoreRpcServerCommandsDefinitions.h]: Ok, cool

**Pull request merged**